bitcoin-core/HWI
# CheckSig changes (#19)

* add checksig and specter devices

* patch FloatOperation

* backwards compatible inteface for enumerate in custom devices

* adopted warning instead of deprecated warn (#12)

* warning, not warn (#13)

* Improvements (#14)

* Remove .python-version with gitignore (because Achow doesn't know how to use pyenv)

* Improve logger names

* Fix ipc

(cherry picked from commit a67d37216567f73220b1f7cd0e1598c508d6d510)

* Dev (#15)

* Add buttons to psbt dialog

* Add device IDs for DIY Jade on M5StickC-Plus

* jade: update jade simulator build in line with current jade master branch

The Jade qumu simulator has been updated to qemu 8, which requires changes to the configure options when building from source.

* Add Vendor/Device IDs for CH340 based DIY Jade devices.

* _base58: Add checksum checking variant of decode

* bitbos02: Use our own base58 implementation

* deps: Add semver, noiseprotocol, and protobuf as explicit deps

semver is required by jadepy and bitbox02
noiseprotocol and protobuf are required by bitbox02

* bitbox02: vendor bitbox02 library

Bring in a copy of the bitbox02 python library. Not yet used.

* deps: Update hidapi to >= 0.14.0

* bitbox02 lib: Use relative imports and our own base58

Modify bitbox02's vendored library to use relative imports.

Also modify to use our own _base58 instead of an external module.

* bitbox02: Replace installed bitbox02 dep with vendored

Use the vendored package instead installing it as an external dependency.

* gui: Remove default passphrase

* flake8: exclude bitbox02_lib

* bitbox02: implement missing encode_check

* common: enforce enum name is a str

* ci: Break dockerfile build caches

* ci: Test against python 3.11

* build: Use debian oldoldstable instead of fixed version

* ci: Consolidate lint tasks

* ci: Use upstream ledger app builder images

* ci: Use python 3.8 for main image

* ci: Update speculos cmake command

* ci: Remove global container

* jade: Remove jade BLE warning

* ci: Add slirp dep for jade emulator

* ci: Patch keepkey googletest to build

* ci: Run dist tests on all versions

* ci, trezor: Set protobuf version to 3.20

* tests: Use h as hardened marker

Bitcoin Core uses h now.

* ci, speculos: install pytesseract as speculos dependency

* tests, jade: specify pc-bios dir to qemu

* tests, ledger: specify nanos as the model

* coldcard: Drop PyCoin usage entirely

* Bump version to 2.3.0 and regenerate setup.py

* qt: Use correct name for xpub display field

* build: Use a separate container for Windows builds

Windows builds with WINE should use a separate container so that builds on architectures that don't have WINE will still work.

* qt: Include '/*' in getkeypool path regex

* qt: Use correct name for p2pkh radio button

* bitbox02: Detect GUI on Windows as not-external gui

sys.stderr will be None for Windows GUI applications.

* build: Detect architecture and use x86_64 naming

* deps: Update libusb1 to 3.0.0

* build: Bump to libusb 1.0.26 for windows

* pyenv: Set python version to just 3.9

No need to be any more specific than the minor version.

* build: Allow building binaries on any python 3.9

* build: Bump *nix release build environment to more recent 3.9

3.9.17 for linux and macOS

* add checksig and specter devices

* patch FloatOperation

* backwards compatible inteface for enumerate in custom devices

* adopted warning instead of deprecated warn (#12)

* warning, not warn (#13)

* Improvements (#14)

* Remove .python-version with gitignore (because Achow doesn't know how to use pyenv)

* Improve logger names

* Fix ipc

(cherry picked from commit a67d37216567f73220b1f7cd0e1598c508d6d510)

* Borrow ipc code from checksighwe

* Update checksighwe settings

* remove code not in HWI

* removed useless code

* restored

* removed code not in HWI

* preserted naming style (no trailing .log)

* do not initialize Decimal from float

---------










* Rebased on upstream (#16)

* Add buttons to psbt dialog

* Add device IDs for DIY Jade on M5StickC-Plus

* jade: update jade simulator build in line with current jade master branch

The Jade qumu simulator has been updated to qemu 8, which requires changes to the configure options when building from source.

* Add Vendor/Device IDs for CH340 based DIY Jade devices.

* _base58: Add checksum checking variant of decode

* bitbos02: Use our own base58 implementation

* deps: Add semver, noiseprotocol, and protobuf as explicit deps

semver is required by jadepy and bitbox02
noiseprotocol and protobuf are required by bitbox02

* bitbox02: vendor bitbox02 library

Bring in a copy of the bitbox02 python library. Not yet used.

* deps: Update hidapi to >= 0.14.0

* bitbox02 lib: Use relative imports and our own base58

Modify bitbox02's vendored library to use relative imports.

Also modify to use our own _base58 instead of an external module.

* bitbox02: Replace installed bitbox02 dep with vendored

Use the vendored package instead installing it as an external dependency.

* gui: Remove default passphrase

* flake8: exclude bitbox02_lib

* bitbox02: implement missing encode_check

* common: enforce enum name is a str

* ci: Break dockerfile build caches

* ci: Test against python 3.11

* build: Use debian oldoldstable instead of fixed version

* ci: Consolidate lint tasks

* ci: Use upstream ledger app builder images

* ci: Use python 3.8 for main image

* ci: Update speculos cmake command

* ci: Remove global container

* jade: Remove jade BLE warning

* ci: Add slirp dep for jade emulator

* ci: Patch keepkey googletest to build

* ci: Run dist tests on all versions

* ci, trezor: Set protobuf version to 3.20

* tests: Use h as hardened marker

Bitcoin Core uses h now.

* ci, speculos: install pytesseract as speculos dependency

* tests, jade: specify pc-bios dir to qemu

* tests, ledger: specify nanos as the model

* coldcard: Drop PyCoin usage entirely

* Bump version to 2.3.0 and regenerate setup.py

* qt: Use correct name for xpub display field

* build: Use a separate container for Windows builds

Windows builds with WINE should use a separate container so that builds on architectures that don't have WINE will still work.

* qt: Include '/*' in getkeypool path regex

* qt: Use correct name for p2pkh radio button

* bitbox02: Detect GUI on Windows as not-external gui

sys.stderr will be None for Windows GUI applications.

* build: Detect architecture and use x86_64 naming

* deps: Update libusb1 to 3.0.0

* build: Bump to libusb 1.0.26 for windows

* pyenv: Set python version to just 3.9

No need to be any more specific than the minor version.

* build: Allow building binaries on any python 3.9

* build: Bump *nix release build environment to more recent 3.9

3.9.17 for linux and macOS

* add checksig and specter devices

* patch FloatOperation

* backwards compatible inteface for enumerate in custom devices

* adopted warning instead of deprecated warn (#12)

* warning, not warn (#13)

* Improvements (#14)

* Remove .python-version with gitignore (because Achow doesn't know how to use pyenv)

* Improve logger names

* Fix ipc

(cherry picked from commit a67d37216567f73220b1f7cd0e1598c508d6d510)

---------










* add checksig and specter devices

---------
